snuba/api.py

```diff
@@ -308,37 +314,34 @@ def format_query(dataset, body, table, prewhere_conditions, final) -> str:
 
 
 @split_query
-def parse_and_run_query(dataset, body, timer):
+def parse_and_run_query(dataset, request: Request, timer):
```

Comment:
Starting to think that `dataset` should actually be an attribute on `request`. I think this can wait a little bit, though.

Reply:
Deferring this for now pending some upcoming changes to datasets.